lib/core/src/Cardano/Pool/DB/Sqlite.hs

```diff
@@ -196,62 +203,104 @@ newDBLayer trace fp = do
                 [ StakeDistributionEpoch ==. fromIntegral epoch ]
                 []
 
-        , putPoolRegistration = \point PoolRegistrationCertificate
-            { poolId
-            , poolOwners
-            , poolMargin
-            , poolCost
-            , poolPledge
-            , poolMetadata
-            } -> do
-            let poolMarginN = fromIntegral $ numerator $ getPercentage poolMargin
-            let poolMarginD = fromIntegral $ denominator $ getPercentage poolMargin
-            let poolCost_ = getQuantity poolCost
-            let poolPledge_ = getQuantity poolPledge
-            let poolMetadataUrl = fst <$> poolMetadata
-            let poolMetadataHash = snd <$> poolMetadata
-            deleteWhere [PoolOwnerPoolId ==. poolId, PoolOwnerSlot ==. point]
-            _ <- repsert (PoolRegistrationKey poolId point) $ PoolRegistration
-                    poolId
-                    point
-                    poolMarginN
-                    poolMarginD
-                    poolCost_
-                    poolPledge_
-                    poolMetadataUrl
-                    poolMetadataHash
-            insertMany_ $ zipWith (PoolOwner poolId point) poolOwners [0..]
+        , putPoolRegistration = \cpt cert -> do
+            let CertificatePublicationTime {slotId, slotInternalIndex} = cpt
+            let poolId = view #poolId cert
+            deleteWhere [PoolOwnerPoolId ==. poolId, PoolOwnerSlot ==. slotId]
```

Comment:
This looks as though it may be deleting _too many_ rows.

We probably want to restrict this to deleting rows with a matching value of `PoolOwnerSlotInternalIndex`.

Reply:
> This looks as though it may be deleting _too many_ rows.
> 
> We probably want to restrict this to deleting rows with a matching value of `PoolOwnerSlotInternalIndex`.

Fixed in 408663203253a2482fb990714f5dfe32250f73cf.